src/python/pants/pantsd/service/pants_service.py

```diff
@@ -205,3 +186,28 @@ def is_terminating(self):
     """
     with self._lock:
       return self._state == self._TERMINATING
+
+
+@frozen_after_init
+@dataclass(unsafe_hash=True)
+class PantsServices:
+  """A registry of PantsServices instances"""
+  services: Tuple[PantsService, ...]
+  port_map: Dict
+  lifecycle_lock: Any
+
+  def __init__(
+    self,
+    services: Optional[Tuple[PantsService, ...]] = None,
+    # A dict of (port_name -> port_info) for named ports hosted by the services.
+    port_map: Optional[Dict] = None,
+    # A lock to guard lifecycle changes for the services. This can be used by individual services
```

Comment:
nit: should these comments be placed near the class members rather than near the arguments to the constructor?

Reply:
Even better, will move to docstring :)